service/terraform-plugin/src/main/java/com/solace/maas/ep/event/management/agent/plugin/terraform/manager/TerraformLogProcessingService.java

```diff
@@ -0,0 +1,168 @@
+        }
+
+        Path out = Files.createTempFile(logPath, System.currentTimeMillis() + "-"
+                + request.getCorrelationId() + "-job", ".log");
+
+        Files.write(out, logs, Charset.defaultCharset());
+    }
+
+    public CommandResult buildTfCommandResult(List<String> jsonLogs) {
+        if (CollectionUtils.isEmpty(jsonLogs)) {
+            throw new IllegalArgumentException("Cannot extract information empty logs");
+        }
+
+        List<Map<String, Object>> successLogs = jsonLogs.stream()
+                .map(this::parseTfOutput)
+                .filter(this::isApplyCompleteLog)
+                .map(this::simplifyApplyCompleteLog)
+                .toList();
+
+        List<Map<String, Object>> errorLogs = jsonLogs.stream()
+                .map(this::parseTfOutput)
+                .filter(this::isErrorLog)
+                .map(this::simplifyApplyErroredLog)
+                .toList();
+        JobStatus status = CollectionUtils.isEmpty(errorLogs) ? JobStatus.success : JobStatus.error;
+        return CommandResult.builder()
+                .logs(successLogs)
+                .errors(errorLogs)
+                .status(status)
+                .build();
+
+    }
+
+
+    private Map<String, Object> parseTfOutput(String json) {
+        try {
+            return objectMapper.readValue(json, Map.class);
+        } catch (Exception e) {
+            throw new IllegalArgumentException("Error parsing JSON: " + json, e);
+        }
+    }
+
+    private String extractValueFromLog(Map<String, Object> map, String key) {
+        return String.valueOf(map.get(key));
+    }
+
+    private Map<String, Object> simplifyApplyCompleteLog(Map<String, Object> expandedLogMessage) {
+        if (!isApplyCompleteLog(expandedLogMessage)) {
+            throw new UnsupportedOperationException(
+                    String.format("This method only handles logs of type %s", VALUE_TYPE_APPLY_COMPLETE)
+            );
+        }
+
+        return Map.of(
+                "address", extractResourceAddressFromHook(expandedLogMessage.get("hook")),
+                "message", expandedLogMessage.get(KEY_MESSAGE)
+        );
+
+    }
+
+    private Map<String, Object> simplifyApplyErroredLog(Map<String, Object> expandedLogMessage) {
+        if (!isErrorLog(expandedLogMessage)) {
+            throw new UnsupportedOperationException(
+                    String.format("This method only handles logs of type %s", VALUE_TYPE_APPLY_ERRORED)
+            );
+        }
+        //return expandedLogMessage;
+        return Map.of(
+                "address", extractResourceAddressFromDiagnostic(expandedLogMessage.get(KEY_DIAGNOSTIC)),
+                KEY_DIAGNOSTIC_DETAIL, extractDiagnosticDetailMessage(expandedLogMessage.get(KEY_DIAGNOSTIC))
+        );
+
+    }
+
+    private String extractDiagnosticDetailMessage(Object diagnostic) {
+        if (diagnostic == null) {
+            return "";
+        }
+        if (!(diagnostic instanceof Map)) {
```

Comment:
Can you use the new java 17 power here? https://mydeveloperplanet.com/2021/09/28/whats-new-between-java-11-and-java-17/

![Screenshot 2023-11-24 at 9 15 01 AM](https://github.com/SolaceProducts/event-management-agent/assets/40175386/25e4e4d4-f6e2-4b31-8d23-77c1c5405959)